tracer/src/Datadog.Trace/Agent/IApiResponse.cs

```diff
@@ -82,5 +96,84 @@ public static bool ShouldRetry(this IApiResponse response)
 
             return shouldRetry;
         }
+
+        /// <summary>
+        /// Gets the <see cref="Encoding"/> represented by the charset defined in the content type header.
+        /// </summary>
+        /// <param name="contentTypeHeader">The raw content-type header, for example <c>"application/json;charset=utf-8"</c></param>
+        /// <returns>The encoding associated with the charset, or <see cref="EncodingHelpers.Utf8NoBom"/> if the content-type header was not provided,
+        /// if the charset was not provided, or if the charset was not recognized</returns>
+        public static Encoding GetCharsetEncoding(string contentTypeHeader)
+        {
+            // special casing application/json because it's so common
+            if (string.IsNullOrEmpty(contentTypeHeader)
+                || string.Equals("application/json", contentTypeHeader, StringComparison.OrdinalIgnoreCase))
+            {
+                // Default
+                return EncodingHelpers.Utf8NoBom;
+            }
+
+            // text/plain; charset=utf-8; boundary=foo
+            foreach (var pair in contentTypeHeader.SplitIntoSpans(';'))
```

Comment:
Can we can skip the first item here.

Reply:
> Can we can skip the first item here

I don't know 😅 

Is `charset=utf-8;text/plain` a valid content-type header? Can we guarantee we won't _receive_ that? Is it worth worrying about? 🤷‍♂️

Reply:
> Is `charset=utf-8;text/plain` a valid `content-type` header?

[No](https://www.rfc-editor.org/rfc/rfc9110#section-8.3.1)

> Is it worth worrying about?

Big "meh." Never mind 😄